test/tests/controls/time_periods/materials/material.i

```diff
@@ -0,0 +1,74 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 10
+  ny = 10
+[]
+
+[Variables]
+  [./u]
+  [../]
+[]
+
+[Kernels]
+  [./diff]
+    type = MatDiffusion
+    variable = u
+    prop_name = 'coef'
+  [../]
+  [./time]
+    type = TimeDerivative
+    variable = u
+  [../]
+[]
+
+[./Materials]
+  [./mat0]
+    type = GenericConstantMaterial
+    prop_names = 'coef'
+    prop_values = '0.1'
+  [../]
+  [./mat1]
+    type = GenericConstantMaterial
+    prop_names = 'coef'
+    prop_values = '0.4'
+  [../]
+[]
+
+[BCs]
+  [./left]
+    type = DirichletBC
+    variable = u
+    boundary = left
+    value = 0
+  [../]
+  [./right]
+    type = DirichletBC
+    variable = u
+    boundary = right
+    value = 1
+  [../]
+[]
+
+[Executioner]
+  type = Transient
+  num_steps = 10
+  dt = 0.1
+  solve_type = PJFNK
+  petsc_options_iname = '-pc_type -pc_hypre_type'
+  petsc_options_value = 'hypre boomeramg'
+[]
+
+[Outputs]
+  exodus = true
+[]
+
+[Controls]
+  [./material]
+    type = TimePeriod
```

Comment:
General comment not related to this PR:
I just thought about the case where a user inverts the start and end time. I think that case should be an error. Right now if you do that, it'll just always permanently do the opposite of what you requested (always disable an "enable_objects" or always enable a "disable_objects").